frontends/web/src/routes/account/add/add.tsx

```diff
@@ -109,29 +100,71 @@ class ManageAccount extends Component<Props, State> {
                 apiPost('account-add', {
                     coinCode,
                     name: accountName,
-                }).then((data: ResponseData) => {
-                    if (data.success) {
-                        this.setState({ step: 2 });
-                        //route('/account/' + data.accountCode);
-                    } else {
-                        if (data.errorCode) {
-                            alertUser(this.props.t(`addAccount.error.${data.errorCode}`));
-                        } else if (data.errorMessage) {
-                            alertUser(this.props.t('unknownError', { errorMessage: data.errorMessage }));
+                })
+                    .then((data: ResponseData) => {
+                        if (data.success) {
+                            this.setState({ step: 'success' });
+                            //route('/account/' + data.accountCode);
+                        } else {
+                            this.showError(data);
                         }
-                    }
-                });
+                    })
+                    .catch(error => this.showError(error));
                 break;
             case 'success':
+                // TODO: route to manage accounts to configure tokens
                 route('/account-summary');
                 return;
         }
     }
 
+    private showError = (error) => {
+        const { t } = this.props;
+        if (!error) {
+            return;
+        }
+        if (error.errorCode) {
+            this.setState({
+                errorMessage: t(`addAccount.error.${error.errorCode}`)
+            });
+        } else if (error.errorMessage) {
+            this.setState({
+                errorMessage: t('unknownError', { errorMessage: error.errorMessage })
+            });
+        } else if (error) {
+            this.setState({
+                errorMessage: error
+            });
+        }
+    }
+
+    private isFirstStep = () => {
+        switch (this.state.step) {
+            case 'select-coin':
+                return true;
+            case 'choose-name':
+                return this.onlyOneSupportedCoin();
+            case 'success':
+                return false;
+        }
+    }
+
+    private focusRef = (ref) => {
+        setTimeout(() => {
+            if (ref === document.activeElement) {
+                return;
+            }
+            if (document.activeElement instanceof HTMLElement) {
+                document.activeElement?.blur();
```

Comment:
why is this needed? does `.focus()` not take care of that?

Reply:
It didn't but I can test again, but I am affraid it will behave differently in the Qt app, common focus issue. BTW I really look forward using react hooks for this.

Reply:
removed and tested with qt-osx, got focus